lfs.c

```diff
@@ -2400,23 +2418,22 @@ int lfs_file_opencfg(lfs_t *lfs, lfs_file_t *file,
     // setup simple file details
     int err;
     file->cfg = cfg;
-    file->flags = flags | LFS_F_OPENED;
     file->pos = 0;
     file->off = 0;
+    file->flags = flags;
     file->cache.buffer = NULL;
 
+    // get id, add to list of mdirs to catch update changes
+    file->type = LFS_TYPE_REG;
+    lfs_mlist_add(lfs, (struct lfs_mlist *)file);
+
```

Comment:
I wouldn't move this either, the `lfs_mdir_t` is not initialized until the `lfs_dir_find`, and I'm not sure having an uninitialized mdir on in mlist is safe.

Unfortunately the mlist is a bit sensitive because all wear-leveling/allocation logic depend on it. It would be better if we prefer changing the LFS_F_OPENED behavior slightly to avoid changing the mlist logic, since LFS_F_OPENED asserts are just protection against already invalid API usage.

Reply:
its honestly better to leave alone the LFS_F_OPENED flag honestly. It causes those problems with the assert and the `isused` check can be utilized only when need (in `file_opencfg` and `dir_open`) and the rest of the API can stay with the flag